upgrade/upgrade-4.1.3.php

```diff
@@ -26,5 +26,7 @@
  */
 function upgrade_module_4_1_3($object)
 {
-    return $object->unregisterHook('displayHome');
+    return $object->unregisterHook('displayHome')
+        && Db::getInstance()->execute('ALTER TABLE `' . _DB_PREFIX_ . 'ganalytics` CHARACTER SET = utf8mb4 COLLATE utf8mb4_general_ci;')
+        && Db::getInstance()->execute('ALTER TABLE `' . _DB_PREFIX_ . 'ganalytics_data` CHARACTER SET = utf8mb4 COLLATE utf8mb4_general_ci;');
```

Comment:
I wonder about this. Should we force the charset? Or should we not do something?

- so people who install ps_googleanalytics for the 1st time will have utf8mb4
- people who upgrade to 4.1.3 will keep their previous charset

I think maybe some people already fixed their charset in database or even chose another one. So forcing, during the upgrade step, the charset is a risky move.

Reply:
Agree. We need keep changes as safe as possible.

Reply:
Hello @matks 

We could check the character set before?
This big change was made in 1.7.7.0 without any verification, this is the recommendation of the Mysql documentation 

https://github.com/PrestaShop/autoupgrade/blob/dev/upgrade/sql/1.7.7.0.sql#L64-L307